types/encoding.go

```diff
@@ -1288,8 +1288,8 @@ func (ren *V2FileContractRenewal) DecodeFrom(d *Decoder) {
 }
 
 // DecodeFrom implements types.DecoderFrom.
-func (fcf V2FileContractFinalization) DecodeFrom(d *Decoder) {
-	(*V2FileContract)(&fcf).DecodeFrom(d)
+func (fcf *V2FileContractFinalization) DecodeFrom(d *Decoder) {
+	(*V2FileContract)(fcf).DecodeFrom(d)
```

Comment:
This fixes the bug found here https://github.com/SiaFoundation/walletd/pull/154/files#r1676487216